### PR TITLE
[JAMES-2806] [Part 1] ObjectStorageBlobsDAO with BucketName

### PR DESCRIPTION
**Pull request by trantienduchn**
This PR is cut from the original #2477 one because It was already to big. This PR contains list of commits had been approved and some small fixup beginning from f7917eb (you cat start to review from it), which are implemented:
 - Put BucketName into BlobStore methods
 - Modify the ObjectStorage Configuration to make namespace become optional + add one more prefix bucket name configuration item.
 - Update ObjectStorageBlobsDAO to work with BucketName passed in argument list

Otherwise it lack some works that will be implemented in other PR:
 - Bucket Name calculation from bucket prefix is not applied yet (I just add the configuration item, just it)
 - Saving blobs behavior need to be update to match with the discussion described in https://github.com/linagora/james-project/pull/2477#discussion_r299281042

### CODE REVIEW

**Comment by trantienduchn** [comment]
```
[876ea8a94eae2f97877d89d8be169a507da0e5fa] [ERROR] Errors: 
[876ea8a94eae2f97877d89d8be169a507da0e5fa] [ERROR]   MemoryTaskManagerTest.getStatusShouldReturnFailedWhenExceptionThrown » Bubbling
```
test this please

File: server/container/guice/blob-objectstorage-guice/src/test/java/org/apache/james/modules/objectstorage/swift/ObjectStorageBlobStoreModuleTest.java [hunk 1]
```diff
@@ -92,11 +90,11 @@ static void setUp(DockerSwift dockerSwift) throws Exception {
                         .build()),
                     Optional.empty(),
                     Optional.empty()))
+                .defaultBucketName(Optional.of(generateBucketName()))
```
Comment (chibenwa) [inline server/container/guice/blob-objectstorage-guice/src/test/java/org/apache/james/modules/objectstorage/swift/ObjectStorageBlobStoreModuleTest.java]: Can be contracted to `.defaultBucketName(generateBucketName())`

**Comment by chibenwa** [comment]
```
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] [ERROR]   LinshareBlobExportMechanismTest.exportShouldShareTheDocumentViaLinshare:90 
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] Expecting all elements of:
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53]   <[org.apache.james.linshare.client.ReceivedShare@186569c8]>
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] to satisfy given requirements, but this element did not:
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53]   <org.apache.james.linshare.client.ReceivedShare@186569c8> 
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] Details: "
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] Expecting:
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53]  <"deleted-message-of-bob&#64;james.org-ed7002b439e9ac845f22357d822bac1444730fbdb6016d3ec9432297b9ec9f73.txt">
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] to start with:
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53]  <"deleted-message-of-bob@james.org-">
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] "
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] [ERROR]   LinshareBlobExportMechanismTest.exportWithFilePrefixShouldCreateFileWithCustomPrefix:167 
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] Expecting:
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53]  <"deleted-message-of-bob&#64;james.orged7002b439e9ac845f22357d822bac1444730fbdb6016d3ec9432297b9ec9f73.txt">
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] to start with:
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53]  <"deleted-message-of-bob@james.org">
```

test this please

**Comment by chibenwa** [comment]
`LinshareBlobExportMechanismTest.exportShouldShareTheDocumentViaLinshare:90`

test this please

**Comment by trantienduchn** [comment]
```
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] [ERROR] Failures: 
[e8d9e249f2e8b9bff21cfe4adce8d77965584b53] [ERROR]   LinshareBlobExportMechanismTest.exportShouldShareTheDocumentViaLinshare:90 
```
test this please

**Comment by trantienduchn** [comment]
test this please

**Comment by trantienduchn** [comment]
rebase to resolve conflict

**Comment by trantienduchn** [comment]
rebase to resolve conflict

**Comment by chibenwa** [comment]
```

[44e310843134c666233113810133a2e252dda8d0] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution
[44e310843134c666233113810133a2e252dda8d0] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution
[44e310843134c666233113810133a2e252dda8d0] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution
[44e310843134c666233113810133a2e252dda8d0] [ERROR]   MailRepositoryContract.storingAndRemovingMessagesConcurrentlyShouldLeadToConsistentResult:474 » Execution
```

test this please

can be considered :green_apple: as subsequent builds are :green_apple:

**Comment by Arsnael** [comment]
```
[44e310843134c666233113810133a2e252dda8d0] [ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:javadoc (create-javadocs) on project apache-james-mpt-core: Execution create-javadocs of goal org.apache.maven.plugins:maven-javadoc-plugin:2.10.4:javadoc failed: Plugin org.apache.maven.plugins:maven-javadoc-plugin:2.10.4 or one of its dependencies could not be resolved: Failed to collect dependencies at org.apache.maven.plugins:maven-javadoc-plugin:jar:2.10.4 -> org.apache.maven:maven-archiver:jar:2.5: Failed to read artifact descriptor for org.apache.maven:maven-archiver:jar:2.5: Could not transfer artifact org.apache.maven:maven-archiver:pom:2.5 from/to central (https://repo.maven.apache.org/maven2): Failed to transfer file: https://repo.maven.apache.org/maven2/org/apache/maven/maven-archiver/2.5/maven-archiver-2.5.pom. Return code is: 503 , ReasonPhrase:Maximum threads for service reached. -> [Help 1]
```
Pfffff... Tests are :green_apple: though, so can be considered :green_apple: 

test this please

**Pull request merged**